### PR TITLE
setting to prevent links opening associated apps

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: https://app.asana.com/0/inbox/392891325557408/1145691603022750/1146067154966667
Tech Design URL:
CC:

**Description**:

Added setting to allow user to disable opening associated apps.

**Steps to test this PR**:
1. Check associated apps open by default (e.g. Twitter, Amazon, Facebook)
1. Change setting and check associated apps not open by default
1. Change setting back and check they open again.
1. Ensure other settings still work as expected.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/TabViewController.swift [hunk 1]
```diff
@@ -874,13 +876,13 @@ extension TabViewController: WKNavigationDelegate {
...
         guard let documentUrl = navigationAction.request.mainDocumentURL else {
-            completion(.allow)
+            completion(allowPolicy)
             return
         }
         
         if shouldReissueSearch(for: url) {
             reissueSearchWithStatsParams(for: url)
-            completion(.cancel)
+            completion(allowPolicy)
```
Comment (bwaresiak) [inline DuckDuckGo/TabViewController.swift]: @brindy This change made ATB test fail, as we now don't reissue search as we used to.

Comment (brindy) [reply DuckDuckGo/TabViewController.swift]: Thanks @bwaresiak - the system is working!! :D

**Review by bwaresiak — COMMENTED** [review]
Awesome to see that in action :)

Two things:
iPhone SE setting layout is broken:

iphse

Next - this is not something new, but prolly should be addressed somehow:
1. Search for some video on SERP (videos tab).
2. Navigate to it.
3. Press back. Video overlay is visible on SERP.

backbutton

**Comment by brindy** [comment]
Thanks - that should fix the layout.

As for that overlay - I'm ok with it if it's not new.  I don't actually see it myself.

**Review by bwaresiak — APPROVED** [review]
Great LGTM!

**Comment by brindy** [comment]
Thanks!

**Pull request merged**